pkgs/development/compilers/solc/default.nix

```diff
@@ -43,89 +45,111 @@ let
   };
 
   pname = "solc";
-  version = "0.8.13";
   meta = with lib; {
     description = "Compiler for Ethereum smart contract language Solidity";
     homepage = "https://github.com/ethereum/solidity";
-    license = licenses.gpl3;
+    license = licenses.gpl3Plus;
     maintainers = with maintainers; [ dbrock akru lionello sifmelcara ];
   };
 
-  solc = if gccStdenv.isLinux then gccStdenv.mkDerivation rec {
-    inherit pname version meta;
-
-    # upstream suggests avoid using archive generated by github
-    src = fetchzip {
-      url = "https://github.com/ethereum/solidity/releases/download/v${version}/solidity_${version}.tar.gz";
-      hash = "sha256-cFC9M65kSYgYq9rhBXZKEdfvIMbMaDiDwdPmU8v9s7k=";
-    };
-
-    postPatch = ''
-      substituteInPlace cmake/jsoncpp.cmake \
-        --replace "${jsoncppUrl}" ${jsoncpp}
-      substituteInPlace cmake/range-v3.cmake \
-        --replace "${range3Url}" ${range3}
-      substituteInPlace cmake/fmtlib.cmake \
-        --replace "${fmtlibUrl}" ${fmtlib}
-    '';
-
-    cmakeFlags = [
-      "-DBoost_USE_STATIC_LIBS=OFF"
-
-    ] ++ (if z3Support then [
-      "-DSTRICT_Z3_VERSION=OFF"
-    ] else [
-      "-DUSE_Z3=OFF"
-    ]) ++ lib.optionals (!cvc4Support) [
-      "-DUSE_CVC4=OFF"
-    ];
-
-    nativeBuildInputs = [ cmake ];
-    buildInputs = [ boost ]
-      ++ lib.optionals z3Support [ z3 ]
-      ++ lib.optionals cvc4Support [ cvc4 cln gmp ];
-    checkInputs = [ ncurses python3 ];
-
-    # tests take 60+ minutes to complete, only run as part of passthru tests
-    doCheck = false;
-
-    checkPhase = ''
-      pushd ..
-      # IPC tests need aleth avaliable, so we disable it
-      sed -i "s/IPC_ENABLED=true/IPC_ENABLED=false\nIPC_FLAGS=\"--no-ipc\"/" ./scripts/tests.sh
-      for i in ./scripts/*.sh ./scripts/*.py ./test/*.sh ./test/*.py; do
-        patchShebangs "$i"
-      done
-      TERM=xterm ./scripts/tests.sh ${if z3Support then "--no-smt" else ""}
-      popd
-    '';
-
-    doInstallCheck = true;
-    installCheckPhase = ''
-      $out/bin/solc --version > /dev/null
-    '';
-
-    passthru.tests = {
-      solcWithTests = solc.overrideAttrs (attrs: { doCheck = true; });
-    };
-  } else gccStdenv.mkDerivation rec {
-    inherit pname version meta;
-
-    src = pkgs.fetchurl {
-      url = "https://github.com/ethereum/solidity/releases/download/v${version}/solc-macos";
-      sha256 = "sha256-FNTvAT6oKtlekf2Um3+nt4JxpIP/GnnEPWzFi4JvW+o=";
-    };
-    dontUnpack = true;
-
-    installPhase = ''
-      runHook preInstall
-
-      mkdir -p $out/bin
-      cp ${src} $out/bin/solc
-      chmod +x $out/bin/solc
-
-      runHook postInstall
-    '';
-  };
+  makeSolc =
+    { version
+    , linuxHash
+    , macosHash
+    }:
+    let
+      solc =
+        if gccStdenv.isLinux then
+          gccStdenv.mkDerivation
+            rec {
+              inherit pname version meta;
+
+              # upstream suggests avoid using archive generated by github
```

Comment:
This only adds jsoncpp https://github.com/ethereum/solidity/blob/develop/scripts/create_source_tarball.sh#L32-L37 which we want to use our own anyway. Also the upstream version is currently out of date.